### DISCUSSION TITLE
Plotting plot_mode_3d

### DISCUSSION THREAD

**Discussion by rahultarale** [petrobras/ross#997] [General]
Hi,

Can you please help me to get the code to plot critical speed using plot_mode_3d. I have done it for modal analysis, but not able to figure out for critical speed.

```python
for mode in range(0,5):
    fig3 = modal.plot_mode_3d(mode,frequency_type='wn',frequency_units='RPM')
    fig10 = modal.plot_orbit(mode,nodes=[0],frequency_type='wn',frequency_units='RPM')
    fig3.show()
    fig10.show()
```

**Comment by raphaeltimbo** [comment]
In rotating equipment the critical frequencies change with speed.
The best way to visualize this is using the Campbell diagram.

```python
import ross as rs
import numpy as np


rotor = rs.rotor_example()
speed = np.linspace(0, 400, 101)
camp = rotor.run_campbell(speed)
camp.plot()
```
![image](https://github.com/petrobras/ross/assets/18506378/40f473c1-d112-448f-9a5f-2b6bda2af588)

You can see that the first forward critical speed is at 96.30 rad/s.

Now you can run a modal analysis at this speed and then plot the mode shape for that critical speed:

```python
modal = rotor.run_modal(speed=96.30)
modal.plot_mode_3d(1)
```
![image](https://github.com/petrobras/ross/assets/18506378/cb2426b8-602b-4d6a-9389-d9d7c639bc76)

**Reply by rahultarale** [reply]
Thanks for the information

**Comment by FadliM21** [comment]
> In rotating equipment the critical frequencies change with speed. The best way to visualize this is using the Campbell diagram.
> 
> ```python
> import ross as rs
> import numpy as np
> 
> 
> rotor = rs.rotor_example()
> speed = np.linspace(0, 400, 101)
> camp = rotor.run_campbell(speed)
> camp.plot()
> ```
> 
> ![image](https://private-user-images.githubusercontent.com/18506378/254071335-40f473c1-d112-448f-9a5f-2b6bda2af588.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIwMTYxMDEsIm5iZiI6MTcxMjAxNTgwMSwicGF0aCI6Ii8xODUwNjM3OC8yNTQwNzEzMzUtNDBmNDczYzEtZDExMi00NDhmLTlhNWYtMmI2YmRhMmFmNTg4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAxVDIzNTY0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMwYTI2YmE3NjNlMjgzYmQwMGZjYmRkM2JjMWRkNzk5NjgzZDMyODVjZjdhNTg2Yzk1NWNmMDU0MmQzNzgyOGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.R3b2_AI1PF7gLYxBoEVMVvTJcO_4yHq92OwyhWnvjMY)
> 
> You can see that the first forward critical speed is at 96.30 rad/s.
> 
> Now you can run a modal analysis at this speed and then plot the mode shape for that critical speed:
> 
> ```python
> modal = rotor.run_modal(speed=96.30)
> modal.plot_mode_3d(1)
> ```
> 
> ![image](https://private-user-images.githubusercontent.com/18506378/254072462-cb2426b8-602b-4d6a-9389-d9d7c639bc76.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIwMTYxMDEsIm5iZiI6MTcxMjAxNTgwMSwicGF0aCI6Ii8xODUwNjM3OC8yNTQwNzI0NjItY2IyNDI2YjgtNjAyYi00ZDZhLTkzODktZDlkN2M2MzliYzc2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAxVDIzNTY0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTljMGNhYjYxZGVjODc1OGZiNzgyNTFjOGViMWFjNjQ1MTM5ODVlZTE0ZTA3YmExYjZmYTdjMDczMmI0ODNmZWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-dZHuQbIkwt-47B_Cu3AMyYB8iJcXznxBE9VOcvR1ko)



> In rotating equipment the critical frequencies change with speed. The best way to visualize this is using the Campbell diagram.
> 
> ```python
> import ross as rs
> import numpy as np
> 
> 
> rotor = rs.rotor_example()
> speed = np.linspace(0, 400, 101)
> camp = rotor.run_campbell(speed)
> camp.plot()
> ```
> 
> ![image](https://private-user-images.githubusercontent.com/18506378/254071335-40f473c1-d112-448f-9a5f-2b6bda2af588.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIwMTYxMDEsIm5iZiI6MTcxMjAxNTgwMSwicGF0aCI6Ii8xODUwNjM3OC8yNTQwNzEzMzUtNDBmNDczYzEtZDExMi00NDhmLTlhNWYtMmI2YmRhMmFmNTg4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAxVDIzNTY0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMwYTI2YmE3NjNlMjgzYmQwMGZjYmRkM2JjMWRkNzk5NjgzZDMyODVjZjdhNTg2Yzk1NWNmMDU0MmQzNzgyOGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.R3b2_AI1PF7gLYxBoEVMVvTJcO_4yHq92OwyhWnvjMY)
> 
> You can see that the first forward critical speed is at 96.30 rad/s.
> 
> Now you can run a modal analysis at this speed and then plot the mode shape for that critical speed:
> 
> ```python
> modal = rotor.run_modal(speed=96.30)
> modal.plot_mode_3d(1)
> ```
> 
> ![image](https://private-user-images.githubusercontent.com/18506378/254072462-cb2426b8-602b-4d6a-9389-d9d7c639bc76.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIwMTYxMDEsIm5iZiI6MTcxMjAxNTgwMSwicGF0aCI6Ii8xODUwNjM3OC8yNTQwNzI0NjItY2IyNDI2YjgtNjAyYi00ZDZhLTkzODktZDlkN2M2MzliYzc2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAxVDIzNTY0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTljMGNhYjYxZGVjODc1OGZiNzgyNTFjOGViMWFjNjQ1MTM5ODVlZTE0ZTA3YmExYjZmYTdjMDczMmI0ODNmZWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-dZHuQbIkwt-47B_Cu3AMyYB8iJcXznxBE9VOcvR1ko)

Hi Raphael,

I'm a beginner for Rotor dynamics. Can you please help me to understand the campbell plot? What is the meaning of Forward, Mixed and Backward in campbell legend?

**Reply by ViniciusTxc3** [reply]
@FadliM21 this topic will be addressed in the #1041, facilitating future discussions.
